tests/foreman/cli/test_remoteexecution.py

```diff
@@ -1019,3 +1020,204 @@ def test_positive_rex_against_infra_hosts(
+        result = rhel_contenthost.execute('systemctl status goferd')
+        assert result.status == 0, 'Failed to start goferd on client'
+
+        # run Ansible rex command to prove ssh provider works, remove katello-agent
+        invocation_command = make_job_invocation(
+            {
+                'job-template': 'Package Action - Ansible Default',
+                'inputs': 'state=absent, name=katello-agent',
+                'search-query': f"name ~ {rhel_contenthost.hostname}",
+            }
+        )
+        result = JobInvocation.info({'id': invocation_command['id']})
+        try:
+            assert result['success'] == '1'
+        except AssertionError:
+            result = 'host output: {}'.format(
+                ' '.join(
+                    JobInvocation.get_output(
+                        {'id': invocation_command['id'], 'host': rhel_contenthost.hostname}
+                    )
+                )
+            )
+            raise AssertionError(result)
+
+        # check katello-agent removal did not influence ygdrassil (SAT-1672)
+        result = rhel_contenthost.execute('yggdrasil status')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        result = rhel_contenthost.execute('systemctl status yggdrasild')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        invocation_command = make_job_invocation(
+            {
+                'job-template': 'Run Command - Script Default',
+                'inputs': 'command=ls',
+                'search-query': f"name ~ {rhel_contenthost.hostname}",
+            }
+        )
+        result = JobInvocation.info({'id': invocation_command['id']})
+        try:
+            assert result['success'] == '1'
+        except AssertionError:
+            result = 'host output: {}'.format(
+                ' '.join(
+                    JobInvocation.get_output(
+                        {'id': invocation_command['id'], 'host': rhel_contenthost.hostname}
+                    )
+                )
+            )
+            raise AssertionError(result)
+
+    @pytest.mark.tier3
+    @pytest.mark.upgrade
+    @pytest.mark.rhel_ver_match('[^6].*')
+    def test_positive_run_job_on_host_registered_to_pull_provider(
+        self,
+        module_org,
+        smart_proxy_location,
+        module_ak_with_cv,
+        module_capsule_configured_mqtt,
+        rhel_contenthost,
+    ):
+        """Run custom template on host registered to mqtt
+
+        :id: 759ad51d-eea7-4d7b-b6ee-60af2b814464
+
+        :expectedresults: Verify the job was successfully ran against the host registered to mqtt
+
+        :CaseImportance: Critical
+
+        :parametrized: yes
+        """
+        result = rhel_contenthost.execute(
+            f'curl -o /etc/pki/ca-trust/source/anchors/satellite-sat-engineering-ca.crt \
+                    {settings.repos["DOGFOOD_REPO_HOST"]}/pub/katello-server-ca.crt; \
```

Comment:
Same thing as above about the potential for this to hide an error.

Reply:
updated, thanks